libnavigation-base/src/main/java/com/mapbox/navigation/base/route/NavigationRoute.kt

```diff
@@ -5,28 +5,192 @@ package com.mapbox.navigation.base.route
 import com.mapbox.api.directions.v5.models.DirectionsResponse
 import com.mapbox.api.directions.v5.models.DirectionsRoute
 import com.mapbox.api.directions.v5.models.DirectionsWaypoint
+import com.mapbox.api.directions.v5.models.LegStep
 import com.mapbox.api.directions.v5.models.RouteLeg
 import com.mapbox.api.directions.v5.models.RouteOptions
-import java.util.UUID
+import com.mapbox.api.directions.v5.models.StepIntersection
+import com.mapbox.api.directions.v5.models.StepManeuver
+import com.mapbox.geojson.LineString
+import com.mapbox.geojson.Point
+import com.mapbox.navigation.base.internal.NativeRouteParserWrapper
+import com.mapbox.navigation.base.internal.SDKRouteParser
+import com.mapbox.navigation.utils.internal.InternalJobControlFactory
+import com.mapbox.navigator.RouteInterface
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.suspendCancellableCoroutine
+import java.net.URL
+import kotlin.coroutines.resume
+import kotlin.coroutines.resumeWithException
 
 /**
  * Wraps a route object used across the Navigation SDK features.
- *
  * @param directionsResponse the original response that returned this route object.
  * @param routeIndex the index of the route that this wrapper tracks
  * from the collection of routes returned in the original response.
  * @param routeOptions options used to generate the [directionsResponse]
- * @param id unique ID of this route entity used for matching it against additional metadata.
- * Unique ID doesn't mean that the underlying [directionsRoute] is unique.
- * Multiple [NavigationRoute]s can represent similar or equal [DirectionsRoute].
  */
-class NavigationRoute(
+class NavigationRoute internal constructor(
```

Comment:
For me it feels that the `NavigationRoute` class has became a way too complex. Isn't is supposed to be a simple DTO which just keeps the data?

Reply:
That was the initial plan, but with the introduction of the native peer, the setup has become more complex.